src/commands/connections/updateWebAppSettings.ts

```diff
@@ -0,0 +1,17 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { WebAppTreeItem } from "src/explorer/WebAppTreeItem";
+import { ext } from "../../extensionVariables";
+
+export async function updateWebAppSetting(webAppId: string, appSettingToUpdate: string, value: string): Promise<void> {
+    const webApp = <WebAppTreeItem | undefined>await ext.tree.findTreeItem(webAppId);
```

Comment:
This is better, but I don't think you quite understood what I meant. You shouldn't need to do `findTreeItem` at all. I used `CosmosDBDatabase` as an example, but it's really `CosmosDBTreeItem.ts` and `ConnectionsTreeItem.ts` where you need to specify the type of the parent. Then you can just do `this.parent.parent.appSettingsNode` within CosmosDBTreeItem.ts